pkg/data/generic.go

```diff
@@ -203,31 +196,81 @@ func (s *GenericSync) sync(resource dynamic.NamespaceableResourceInterface, quit
 }
 
 func (s *GenericSync) syncAdd(obj runtime.Object) error {
-	m, err := meta.Accessor(obj)
+	path, err := s.objPath(obj)
 	if err != nil {
 		return err
 	}
-	name := m.GetName()
-	var path = m.GetName()
-	if s.ns.Namespaced {
-		path = m.GetNamespace() + "/" + name
-	}
 	return s.opa.PutData(path, obj)
 }
 
 func (s *GenericSync) syncRemove(obj runtime.Object) error {
-	m, err := meta.Accessor(obj)
+	path, err := s.objPath(obj)
+	if err != nil {
+		return err
+	}
+	return s.opa.PatchData(path, opa_client.DataPatchOpRemove, nil)
+}
+
+func (s *GenericSync) syncAll(objs []unstructured.Unstructured) error {
+
+	// Build a list of patches to apply.
+	var patches []opa_client.DataPatch
+
+	createdSubPaths := map[string]struct{}{}
+
+	for _, obj := range objs {
+
+		path, err := s.objPath(&obj)
+		if err != nil {
+			return err
+		}
+
+		var subPath string
+		for _, segment := range strings.Split(path, "/") {
+			subPath = filepath.Join(subPath, segment)
```

Comment:
Nit: Will this give us Windows paths? Will that cause problems for unit tests?

Reply:
Good catch! We probably shouldn't use `filepath.Join`, should be `path.Join`.